src/pal/src/thread/thread.cpp

```diff
@@ -287,37 +235,7 @@ static void FreeTHREAD(CPalThread *pThread)
     memset((void*)pThread, 0xcc, sizeof(*pThread));
 #endif
 
-    // We SHOULD be doing the following, but it causes massive problems. See the
-    // comment below.
-    //pthread_setspecific(thObjKey, NULL); // Make sure any TLS entry is removed.
-
-    //
-    // Never actually free the THREAD structure to make the TLS lookaside cache work.
```

Comment:
The TLS lookaside cache that this is talking about was deleted from PAL long time ago.